src/Avalonia.Controls/Platform/ITopLevelImpl.cs

```diff
@@ -80,6 +80,11 @@ public interface ITopLevelImpl : IDisposable
         /// </summary>
         void SetInputRoot(IInputRoot inputRoot);
 
+        /// <summary>
+        /// Gets the <see cref="IInputRoot"/> for the toplevel.
+        /// </summary>
+        IInputRoot InputRoot { get; }
```

Comment:
This changes the platform backend interface, so it involves every backend.

If we are doing that, might as well make it an auto property instead of SetInputRoot + property.

@grokys

Reply:
Done. As for me it is good solution, even it was already done so for `WindowBaseImpl` & used in `WindowBaseEvents` for ex.